routers/repo/setting.go

```diff
@@ -349,6 +349,38 @@ func SettingsPost(ctx *context.Context, form auth.RepoSettingForm) {
 		ctx.Flash.Success(ctx.Tr("repo.settings.wiki_deletion_success"))
 		ctx.Redirect(ctx.Repo.RepoLink + "/settings")
 
+	case "archive":
```

Comment:
I know it would duplicate some part of the code but code would be more readable if there would be separate case for "unarchive"